### PR TITLE
Check the domain in METADATA_SERVICE_ADDR to confirm on EC2 platform

### PR DESCRIPTION
**Pull request by baskaran-md**
Fixes https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/issues/182

### CODE REVIEW

**Comment by baskaran-md** [comment]
I signed it!

File: lib/fluent/plugin/out_google_cloud.rb [hunk 1]
```diff
@@ -768,6 +768,13 @@ def detect_platform
             return Platform::EC2
           end
         end
+        open('http://' + METADATA_SERVICE_ADDR +
+                '/latest/meta-data/services/domain') do |f|
```
Comment (igorpeshansky) [inline lib/fluent/plugin/out_google_cloud.rb]: Should we just be checking that the `/latest` path doesn't return a 404?

Comment (baskaran-md) [reply lib/fluent/plugin/out_google_cloud.rb]: Checking just the `/latest` seems so generic to confirm the EC2 platform.
Even Google / Azure can add that path in future.

**Comment by baskaran-md** [comment]
@igorpeshansky  Yes, the "Server" response header is not present in the latest instance type. 
More details @ https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/issues/182

**Comment by baskaran-md** [comment]
Closing the request as per https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/issues/182#issuecomment-354341343

**Pull request closed without merging**